packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -649,7 +649,7 @@ class _FlutterPlatform extends PlatformPlugin {
     assert(controllerSinkClosed);
     if (outOfBandError != null) {
       printTrace('test $ourTestCount: finished with out-of-band failure');
-      throw outOfBandError;
+      return outOfBandError;
```

Comment:
this seems very unidiomatic. Normally we throw (which in an async function causes the Future to complete with an error) when we have an error.